jdisc_http_service/src/main/java/com/yahoo/jdisc/http/ssl/DefaultSslContextFactoryProvider.java

```diff
@@ -58,20 +62,61 @@ public SslContextFactory getInstance(String containerId, int port) {
                 .toArray(String[]::new);
         factory.setExcludeCipherSuites(excludedCiphersWithoutTlsRsaExclusion);
 
-        setStringArrayParameter(
-                factory, sslConfig.excludeProtocol(), ConnectorConfig.Ssl.ExcludeProtocol::name, SslContextFactory::setExcludeProtocols);
-        setStringArrayParameter(
-                factory, sslConfig.includeProtocol(), ConnectorConfig.Ssl.IncludeProtocol::name, SslContextFactory::setIncludeProtocols);
-        setStringArrayParameter(
-                factory, sslConfig.excludeCipherSuite(), ConnectorConfig.Ssl.ExcludeCipherSuite::name, SslContextFactory::setExcludeCipherSuites);
-        setStringArrayParameter(
-                factory, sslConfig.includeCipherSuite(), ConnectorConfig.Ssl.IncludeCipherSuite::name, SslContextFactory::setIncludeCipherSuites);
-
-        factory.setKeyManagerFactoryAlgorithm(sslConfig.sslKeyManagerFactoryAlgorithm());
-        factory.setProtocol(sslConfig.protocol());
+        // Check if using new ssl syntax from services.xml
+        if (!sslConfig.privateKeyFile().isEmpty()) {
+            factory.setKeyStore(createKeystore(sslConfig));
+            if (!sslConfig.caCertificateFile().isEmpty()) {
+                factory.setTrustStore(createTruststore(sslConfig));
+            }
+            factory.setProtocol("TLS");
+        } else {
+            // TODO Remove SslKeyStoreConfigurator / SslTrustStoreConfigurator
```

Comment:
TODO Vespa 7?

Reply:
TODO later this sprint. This was never a public api :)